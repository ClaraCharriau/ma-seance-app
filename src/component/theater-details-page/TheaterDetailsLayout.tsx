import { Await, Outlet, useLoaderData } from 'react-router-dom';
import style from './TheaterDetailsLayout.module.css';
import FavoriteButton from '../common/favorite-button/FavoriteButton';
import WeekNavigation from '../common/week-navigation/WeekNavigation';
import ErrorPage from '../error-page/Error';
import Spinner from '../common/spinner/Spinner';
import { Theater } from '../../model/Theater';
import { Suspense } from 'react';

const TheaterDetailsLayout = () => {
    const data = useLoaderData() as {
        theater: Theater;
    };

    const IMG_PATH = '/assets/img/theaters';
    const IMG_SUFFIX = '.jpg';

    return (
        <Suspense fallback={<Spinner />}>
            <Await resolve={data.theater} errorElement={<ErrorPage />}>
                {theater => (
                    <>
                        <div className={style.backgroundWrapper}>
                            <img
                                className={style.theaterBackground}
                                src={IMG_PATH + theater.imagePath + IMG_SUFFIX}
                                alt={'photo de ' + theater.name}
                            />
                            <div className={style.fading}></div>
                        </div>

                        <main className={style.theaterDetailsMain}>
                            <div className={style.theaterDetailsHeader}>
                                <div className={style.theaterNameAddress}>
                                    <h2>{theater.name}</h2>
                                    <p>{theater.address}</p>
                                </div>
                                <FavoriteButton itemId={theater.id} itemType="theater" />
                            </div>

                            <section>
                                <h3 className={style.plannerTitle}>Prochaines s√©ances :</h3>
                                <WeekNavigation context="theaters" />
                                <Outlet />
                            </section>
                        </main>
                    </>
                )}
            </Await>
        </Suspense>
    );
};

export default TheaterDetailsLayout;
