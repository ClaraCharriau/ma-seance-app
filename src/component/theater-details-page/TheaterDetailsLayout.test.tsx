import { render, waitFor } from '@testing-library/react';
import TheaterDetailsLayout from './TheaterDetailsLayout';
import { act } from 'react-dom/test-utils';
import { BrowserRouter } from 'react-router-dom';

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useLoaderData: () => {
        return {
            theater: {
                id: 1,
                name: 'UGC Normandie',
                address: '116 bis Av. des Champs-Élysées 75008 Paris',
                imgPath: '/ugc-normandie-paris'
            }
        };
    }
}));

describe('Theater Details Component', () => {
    jest.useFakeTimers().setSystemTime(new Date('2020-01-01'));
    it('renders theater details component', () => {
        // Given
        let component: any;

        // When
        act(() => {
            component = render(
                <BrowserRouter>
                    <TheaterDetailsLayout />
                </BrowserRouter>
            );
        });

        // Then
        waitFor(() => {
            expect(component.container).toMatchSnapshot();
        });
    });
});
