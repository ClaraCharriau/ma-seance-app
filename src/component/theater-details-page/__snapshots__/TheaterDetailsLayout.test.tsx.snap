// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theater Details Component renders theater details component 1`] = `
<div>
  <div
    class="backgroundWrapper"
  >
    <img
      alt="photo de UGC Normandie"
      class="theaterBackground"
      src="/assets/img/theaters/ugc-normandie-paris.jpg"
    />
    <div
      class="fading"
    />
  </div>
  <main
    class="theaterDetailsMain"
  >
    <div
      class="theaterDetailsHeader"
    >
      <div
        class="theaterNameAddress"
      >
        <h2>
          UGC Normandie
        </h2>
        <p>
          116 bis Av. des Champs-Élysées 75008 Paris
        </p>
      </div>
      <button
        class="favoriteButtonWrapper"
        data-testid="button"
        id="favorite-button"
      >
        <svg
          class="heart"
          viewBox="0 0 33 28"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M30.0569 3.26386L30.0588 3.26597C32.9452 6.45492 32.7315 11.7267 29.5079 14.9865L29.5071 14.9872L17.7552 26.9219L17.7534 26.9237C17.5851 27.0964 17.3845 27.2337 17.1633 27.3278C16.9422 27.4218 16.7048 27.4708 16.4648 27.472C16.2249 27.4732 15.987 27.4265 15.765 27.3347C15.5429 27.2428 15.3409 27.1074 15.1709 26.9364L15.1691 26.9346L2.99284 14.8095L2.99236 14.809C2.19548 14.0176 1.56732 13.0709 1.14646 12.0267C0.725595 10.9825 0.520911 9.86282 0.544994 8.73623C0.569076 7.60964 0.821415 6.49993 1.28649 5.47497C1.75157 4.45001 2.41956 3.53143 3.24949 2.77531L3.24949 2.77532L3.25121 2.77373C6.41871 -0.139405 11.6557 0.087266 14.8974 3.32917L14.8974 3.32919L14.9006 3.33229L15.9994 4.41195L16.3538 4.76016L16.7043 4.40797L18.0972 3.00815L18.0981 3.00717C18.8883 2.2087 19.8304 1.57901 20.8681 1.1557C21.9055 0.732453 23.0172 0.524181 24.1364 0.543286C25.2556 0.564788 26.3585 0.817247 27.3768 1.2851C28.3953 1.75303 29.3075 2.42649 30.0569 3.26386Z"
          />
        </svg>
        <p
          class="favoriteButtonText"
        >
          Ce 
          cinéma
           
          n’est pas
           dans
           
          vos favoris
        </p>
      </button>
    </div>
    <section>
      <h3
        class="plannerTitle"
      >
        Prochaines séances :
      </h3>
      <div
        class="shadowWrapper"
      >
        <nav
          class="plannerWrapper"
        >
          <a
            class="dayButton"
            href="/theaters/undefined/day-1"
          >
            <span
              class="dayOfWeek"
            >
              mercredi
            </span>
            <span
              class="dayNumber"
            >
              01
            </span>
            <span
              class="month"
            >
              janvier
            </span>
          </a>
          <a
            class="dayButton"
            href="/theaters/undefined/day-2"
          >
            <span
              class="dayOfWeek"
            >
              jeudi
            </span>
            <span
              class="dayNumber"
            >
              02
            </span>
            <span
              class="month"
            >
              janvier
            </span>
          </a>
          <a
            class="dayButton"
            href="/theaters/undefined/day-3"
          >
            <span
              class="dayOfWeek"
            >
              vendredi
            </span>
            <span
              class="dayNumber"
            >
              03
            </span>
            <span
              class="month"
            >
              janvier
            </span>
          </a>
          <a
            class="dayButton"
            href="/theaters/undefined/day-4"
          >
            <span
              class="dayOfWeek"
            >
              samedi
            </span>
            <span
              class="dayNumber"
            >
              04
            </span>
            <span
              class="month"
            >
              janvier
            </span>
          </a>
          <a
            class="dayButton"
            href="/theaters/undefined/day-5"
          >
            <span
              class="dayOfWeek"
            >
              dimanche
            </span>
            <span
              class="dayNumber"
            >
              05
            </span>
            <span
              class="month"
            >
              janvier
            </span>
          </a>
          <a
            class="dayButton"
            href="/theaters/undefined/day-6"
          >
            <span
              class="dayOfWeek"
            >
              lundi
            </span>
            <span
              class="dayNumber"
            >
              06
            </span>
            <span
              class="month"
            >
              janvier
            </span>
          </a>
          <a
            class="dayButton"
            href="/theaters/undefined/day-7"
          >
            <span
              class="dayOfWeek"
            >
              mardi
            </span>
            <span
              class="dayNumber"
            >
              07
            </span>
            <span
              class="month"
            >
              janvier
            </span>
          </a>
          <a
            class="dayButton"
            href="/theaters/undefined/day-8"
          >
            <span
              class="dayOfWeek"
            >
              mercredi
            </span>
            <span
              class="dayNumber"
            >
              08
            </span>
            <span
              class="month"
            >
              janvier
            </span>
          </a>
          <a
            class="dayButton"
            href="/theaters/undefined/day-9"
          >
            <span
              class="dayOfWeek"
            >
              jeudi
            </span>
            <span
              class="dayNumber"
            >
              09
            </span>
            <span
              class="month"
            >
              janvier
            </span>
          </a>
        </nav>
        <div
          class="shadow"
        />
      </div>
    </section>
  </main>
</div>
`;
